Editor/Data/Attribute/RestrictedAttributeDrawer.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Zinnia.Data.Attribute
+{
+    using UnityEngine;
+    using UnityEditor;
+
+    [CustomPropertyDrawer(typeof(RestrictedAttribute))]
+    public class RestrictedAttributeDrawer : PropertyDrawer
+    {
+        public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+        {
+            EditorGUI.BeginProperty(position, label, property);
+            RestrictedAttribute attrib = attribute as RestrictedAttribute;
+
+            bool playingAndActiveAndEnabled = Application.isPlaying && property.serializedObject.targetObject is Behaviour behaviour && behaviour.isActiveAndEnabled;
+            bool makeReadOnly = (attrib.restrictions & RestrictedAttribute.Restrictions.ReadOnlyAlways) != 0 ||
+                ((attrib.restrictions & RestrictedAttribute.Restrictions.ReadOnlyWhenPlayingAndDisabled) != 0 && playingAndActiveAndEnabled);
+
+            bool muteProperty = (attrib.restrictions & RestrictedAttribute.Restrictions.Muted) != 0;
+
+            bool originalGUIEnabledState = GUI.enabled;
+            if (makeReadOnly)
+            {
+                GUI.enabled = false;
+            }
+
+            Color controlColor = Color.grey;
+
+            Color currentGUIColor = GUI.color;
```

Comment:
I suggest to rename all of these `current*` variables to `original*` like the other one above.